src/BasicVault.sol

```diff
@@ -0,0 +1,60 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity ^0.8.13;
+
+import "yield-utils-v2/token/IERC20.sol";
+
+/**
+@title An on-chain name registry
+@author Calnix
+@dev Vault for a specific ERC20 token; token address to be passed to constructor.
+@notice Vault for a pre-defined ERC20 token that was set on deployment. 
+*/
+
+contract BasicVault {
+    
+    ///@notice ERC20 interface specifying token contract functions
+    ///@dev For constant variables, the value has to be fixed at compile-time, while for immutable, it can still be assigned at construction time.
+    IERC20 public immutable wmdToken;    
+
+    ///@notice mapping addresses to their respective token balances
+    mapping(address => uint) public balances;
+
+    /// @notice Emit event when tokens are deposited into Vault
+    event Deposit(address indexed, uint amt);
+    
+    /// @notice Emit event when tokens are withdrawn from Vault
+    event Withdrawal(address indexed, uint amt);
+
+    constructor(address wmdToken_){
```

Comment:
You can make the parameter of the constructor as `IERC20` this will save the explicit conversion that you are doing.

Reply:
The parameter is only considered an IERC20 in the scope of the compiler. Once the bytecode is generated, the parameter will be an address for purposes of type checking.

Reply:
Meaning, if you have another contract that calls this constructor, and you compile both together, they both need to use IERC20. If you compile them separately and deploy them, one can use IERC20 and the other an address.